server/src/main/java/org/elasticsearch/search/aggregations/bucket/composite/CompositeAggregationBuilder.java

```diff
@@ -154,9 +154,9 @@ public int size() {
     @Override
     protected AggregatorFactory<?> doBuild(SearchContext context, AggregatorFactory<?> parent,
                                            AggregatorFactories.Builder subfactoriesBuilder) throws IOException {
-        if (parent != null) {
+        /*if (parent != null) {
             throw new IllegalArgumentException("[composite] aggregation cannot be used with a parent aggregation");
-        }
+        }*/
```

Comment:
We cannot accept any parent here so as explained in https://github.com/elastic/elasticsearch/issues/28611 we should check that all parent are instances of NestedAggregatorFactory. Something like:
````
private boolean checkParentIsNullOrNested(AggregatorFactory<?> parent) {
        if (parent == null) {
            return true;
        } else if (parent instanceof ReverseNestedAggregatorFactory) {
            return checkParentIsNullOrNested(parent.getParent());
        } else {
            return false;
        }
    }
````
Then we need a test that checks that nested aggregation and composite aggregation works fine together.  `org.elasticsearch.search.aggregations.bucket.NestedIT` would be a good place to write such tests.